# MP2 Part 1: Commit retrieval (rkabarwa)
Retrieves all commits for my 10 assigned projects from World of Code and basic repository info from GitHub.

In [1]:
# !python -m pip install -U python-woc pandas matplotlib tqdm requests tabulate
import json, os, time
import pandas as pd
import requests
from tqdm import tqdm
from woc.remote import WocMapsRemote

woc = WocMapsRemote(base_url="https://worldofcode.org/api/")
netid = 'rkabarwa'

projects = pd.read_csv('net2prj.csv')
projects = projects[projects.netID == netid].reset_index(drop=True)
projects

,netID,WoC,GH
0,rkabarwa,dlr-rm_blenderproc,https://github.com/DLR-RM/BlenderProc
1,rkabarwa,riatelab_cartography,https://github.com/riatelab/cartography
2,rkabarwa,patrickzib_sfa,https://github.com/patrickzib/SFA
3,rkabarwa,openexoplanetcatalogue_open_exoplanet_catalogue,https://github.com/OpenExoplanetCatalogue/open...
4,rkabarwa,snakemake-workflows_rna-seq-kallisto-sleuth,https://github.com/snakemake-workflows/rna-seq...
5,rkabarwa,brianmanderson_dicom_utilities,https://github.com/brianmanderson/dicom_utilities
6,rkabarwa,iraf-community_iraf,https://github.com/iraf-community/iraf
7,rkabarwa,dgerosa_precession,https://github.com/dgerosa/precession
8,rkabarwa,plasmapy_plasmapy,https://github.com/PlasmaPy/PlasmaPy
9,rkabarwa,iqusoft_intel-qs,https://github.com/iqusoft/intel-qs


## Step 1: commit sha1s for each project (p2c)

In [2]:
dfs = []
for prj in projects.WoC:
    shas = woc.get_values('p2c', prj)
    print(prj, len(shas))
    dfs.append(pd.DataFrame({'sha1': shas, 'project': prj}))
    time.sleep(1)
df = pd.concat(dfs, ignore_index=True)
df.to_csv('df_commits.csv', index=False)
print('total', len(df))

dlr-rm_blenderproc 5512


riatelab_cartography 737


patrickzib_sfa 468


openexoplanetcatalogue_open_exoplanet_catalogue 7241


snakemake-workflows_rna-seq-kallisto-sleuth 1421


brianmanderson_dicom_utilities 407


iraf-community_iraf 2783


dgerosa_precession 1191


plasmapy_plasmapy 25704


iqusoft_intel-qs 543


total 46007


## Step 2: commit content in batches of 40 (under the 50 limit), 1s wait between requests
Results are cached in `commit_cache.jsonl` so the cell can be re-run after an interruption without re-downloading.

In [3]:
CACHE = 'commit_cache.jsonl'
done = {}
if os.path.exists(CACHE):
    with open(CACHE) as f:
        for line in f:
            r = json.loads(line)
            done[r['commit']] = r

todo = [s for s in df.sha1.unique() if s not in done]
chunks = [todo[i:i+40] for i in range(0, len(todo), 40)]
missing = {}
with open(CACHE, 'a') as f:
    for chunk in tqdm(chunks):
        for attempt in range(10):
            try:
                res, err = woc.get_values_many('commit.tch', chunk)
                break
            except Exception as e:
                print('retrying after error:', e)
                time.sleep(65)  # WoC asks clients to wait 60s when rate limited
        else:
            raise RuntimeError('batch failed 10 times')
        missing.update(err or {})
        for sha, v in res.items():
            c = v[0]  # [tree, parents, [author, time, tz], [committer, time, tz], message]
            r = {'commit': sha, 'author': c[2][0], 'time': int(c[2][1]), 'message': c[4]}
            done[sha] = r
            f.write(json.dumps(r) + '\n')
        time.sleep(1)

print('retrieved', len(done), 'of', df.sha1.nunique(), 'unique commits;',
      len(missing), 'not found in WoC commit.tch in this run')

  0%|          | 0/1135 [00:00<?, ?it/s]

  0%|          | 1/1135 [00:01<27:12,  1.44s/it]

  0%|          | 2/1135 [00:02<25:16,  1.34s/it]

  0%|          | 3/1135 [00:03<23:56,  1.27s/it]

  0%|          | 4/1135 [00:05<23:29,  1.25s/it]

  0%|          | 5/1135 [00:06<23:04,  1.23s/it]

  1%|          | 6/1135 [00:07<22:54,  1.22s/it]

  1%|          | 7/1135 [00:08<22:29,  1.20s/it]

  1%|          | 8/1135 [00:09<22:19,  1.19s/it]

  1%|          | 9/1135 [00:11<22:25,  1.19s/it]

  1%|          | 10/1135 [00:12<22:13,  1.19s/it]

  1%|          | 11/1135 [00:13<22:14,  1.19s/it]

  1%|          | 12/1135 [00:14<22:19,  1.19s/it]

  1%|          | 13/1135 [00:15<22:16,  1.19s/it]

  1%|          | 14/1135 [00:16<22:17,  1.19s/it]

  1%|▏         | 15/1135 [00:18<22:30,  1.21s/it]

  1%|▏         | 16/1135 [00:19<22:27,  1.20s/it]

  1%|▏         | 17/1135 [00:20<22:23,  1.20s/it]

  2%|▏         | 18/1135 [00:21<22:47,  1.22s/it]

  2%|▏         | 19/1135 [00:23<22:37,  1.22s/it]

  2%|▏         | 20/1135 [00:24<22:30,  1.21s/it]

  2%|▏         | 21/1135 [00:25<22:14,  1.20s/it]

  2%|▏         | 22/1135 [00:26<22:02,  1.19s/it]

  2%|▏         | 23/1135 [00:27<21:48,  1.18s/it]

  2%|▏         | 24/1135 [00:28<21:52,  1.18s/it]

  2%|▏         | 25/1135 [00:30<22:37,  1.22s/it]

  2%|▏         | 26/1135 [00:31<22:19,  1.21s/it]

  2%|▏         | 27/1135 [00:32<22:03,  1.19s/it]

  2%|▏         | 28/1135 [00:33<21:54,  1.19s/it]

  3%|▎         | 29/1135 [00:34<22:01,  1.19s/it]

  3%|▎         | 30/1135 [00:36<21:51,  1.19s/it]

  3%|▎         | 31/1135 [00:37<21:38,  1.18s/it]

  3%|▎         | 32/1135 [00:38<21:34,  1.17s/it]

  3%|▎         | 33/1135 [00:39<21:51,  1.19s/it]

  3%|▎         | 34/1135 [00:40<21:52,  1.19s/it]

  3%|▎         | 35/1135 [00:42<21:40,  1.18s/it]

  3%|▎         | 36/1135 [00:43<21:31,  1.18s/it]

retrying after error: 


  3%|▎         | 37/1135 [02:19<9:03:50, 29.72s/it]

  3%|▎         | 38/1135 [02:20<6:26:40, 21.15s/it]

  3%|▎         | 39/1135 [02:21<4:36:50, 15.16s/it]

  4%|▎         | 40/1135 [02:23<3:19:54, 10.95s/it]

  4%|▎         | 41/1135 [02:24<2:26:10,  8.02s/it]

  4%|▎         | 42/1135 [02:25<1:48:34,  5.96s/it]

  4%|▍         | 43/1135 [02:26<1:22:14,  4.52s/it]

  4%|▍         | 44/1135 [02:27<1:04:10,  3.53s/it]

  4%|▍         | 45/1135 [02:28<51:15,  2.82s/it]  

  4%|▍         | 46/1135 [02:30<42:09,  2.32s/it]

  4%|▍         | 47/1135 [02:31<36:07,  1.99s/it]

  4%|▍         | 48/1135 [02:32<31:37,  1.75s/it]

  4%|▍         | 49/1135 [02:33<28:40,  1.58s/it]

  4%|▍         | 50/1135 [02:34<26:20,  1.46s/it]

  4%|▍         | 51/1135 [02:35<24:41,  1.37s/it]

  5%|▍         | 52/1135 [02:37<23:42,  1.31s/it]

  5%|▍         | 53/1135 [02:38<23:18,  1.29s/it]

  5%|▍         | 54/1135 [02:39<22:30,  1.25s/it]

  5%|▍         | 55/1135 [02:40<22:02,  1.22s/it]

  5%|▍         | 56/1135 [02:41<21:39,  1.20s/it]

  5%|▌         | 57/1135 [02:43<21:27,  1.19s/it]

  5%|▌         | 58/1135 [02:44<21:55,  1.22s/it]

  5%|▌         | 59/1135 [02:45<21:29,  1.20s/it]

  5%|▌         | 60/1135 [02:46<21:41,  1.21s/it]

  5%|▌         | 61/1135 [02:47<21:21,  1.19s/it]

  5%|▌         | 62/1135 [02:49<21:11,  1.19s/it]

  6%|▌         | 63/1135 [02:50<21:26,  1.20s/it]

  6%|▌         | 64/1135 [02:51<21:10,  1.19s/it]

  6%|▌         | 65/1135 [02:52<20:59,  1.18s/it]

  6%|▌         | 66/1135 [02:53<20:56,  1.18s/it]

  6%|▌         | 67/1135 [02:54<21:20,  1.20s/it]

  6%|▌         | 68/1135 [02:56<24:52,  1.40s/it]

  6%|▌         | 69/1135 [02:59<33:25,  1.88s/it]

  6%|▌         | 70/1135 [03:01<29:39,  1.67s/it]

  6%|▋         | 71/1135 [03:02<27:31,  1.55s/it]

  6%|▋         | 72/1135 [03:03<25:57,  1.47s/it]

  6%|▋         | 73/1135 [03:04<24:21,  1.38s/it]

retrying after error: 


  7%|▋         | 74/1135 [04:41<8:48:57, 29.91s/it]

  7%|▋         | 75/1135 [04:42<6:16:29, 21.31s/it]

  7%|▋         | 76/1135 [04:43<4:29:32, 15.27s/it]

  7%|▋         | 77/1135 [04:44<3:14:46, 11.05s/it]

  7%|▋         | 78/1135 [04:46<2:22:57,  8.11s/it]

  7%|▋         | 79/1135 [04:47<1:46:11,  6.03s/it]

  7%|▋         | 80/1135 [04:48<1:20:23,  4.57s/it]

  7%|▋         | 81/1135 [04:49<1:02:32,  3.56s/it]

  7%|▋         | 82/1135 [04:50<50:10,  2.86s/it]  

  7%|▋         | 83/1135 [04:52<41:16,  2.35s/it]

  7%|▋         | 84/1135 [04:53<35:10,  2.01s/it]

  7%|▋         | 85/1135 [04:54<31:14,  1.78s/it]

  8%|▊         | 86/1135 [04:55<27:55,  1.60s/it]

  8%|▊         | 87/1135 [04:56<26:00,  1.49s/it]

  8%|▊         | 88/1135 [04:58<24:11,  1.39s/it]

  8%|▊         | 89/1135 [04:59<23:15,  1.33s/it]

  8%|▊         | 90/1135 [05:00<22:27,  1.29s/it]

  8%|▊         | 91/1135 [05:01<23:29,  1.35s/it]

  8%|▊         | 92/1135 [05:03<22:26,  1.29s/it]

  8%|▊         | 93/1135 [05:04<22:00,  1.27s/it]

  8%|▊         | 94/1135 [05:05<23:01,  1.33s/it]

  8%|▊         | 95/1135 [05:07<22:34,  1.30s/it]

  8%|▊         | 96/1135 [05:08<22:45,  1.31s/it]

  9%|▊         | 97/1135 [05:10<24:18,  1.41s/it]

  9%|▊         | 98/1135 [05:11<23:06,  1.34s/it]

  9%|▊         | 99/1135 [05:12<22:09,  1.28s/it]

  9%|▉         | 100/1135 [05:13<21:41,  1.26s/it]

  9%|▉         | 101/1135 [05:14<21:36,  1.25s/it]

  9%|▉         | 102/1135 [05:16<22:21,  1.30s/it]

  9%|▉         | 103/1135 [05:17<23:26,  1.36s/it]

  9%|▉         | 104/1135 [05:18<22:56,  1.34s/it]

  9%|▉         | 105/1135 [05:20<22:26,  1.31s/it]

  9%|▉         | 106/1135 [05:21<21:43,  1.27s/it]

  9%|▉         | 107/1135 [05:22<21:16,  1.24s/it]

 10%|▉         | 108/1135 [05:23<20:50,  1.22s/it]

 10%|▉         | 109/1135 [05:25<21:16,  1.24s/it]

 10%|▉         | 110/1135 [05:26<20:57,  1.23s/it]

 10%|▉         | 111/1135 [05:28<24:34,  1.44s/it]

 10%|▉         | 112/1135 [05:29<23:49,  1.40s/it]

 10%|▉         | 113/1135 [05:30<23:02,  1.35s/it]

 10%|█         | 114/1135 [05:31<22:25,  1.32s/it]

 10%|█         | 115/1135 [05:33<21:59,  1.29s/it]

 10%|█         | 116/1135 [05:34<21:26,  1.26s/it]

 10%|█         | 117/1135 [05:35<20:54,  1.23s/it]

 10%|█         | 118/1135 [05:36<20:51,  1.23s/it]

 10%|█         | 119/1135 [05:38<21:06,  1.25s/it]

 11%|█         | 120/1135 [05:39<21:11,  1.25s/it]

 11%|█         | 121/1135 [05:40<20:51,  1.23s/it]

 11%|█         | 122/1135 [05:41<20:43,  1.23s/it]

 11%|█         | 123/1135 [05:42<20:49,  1.23s/it]

 11%|█         | 124/1135 [05:44<20:29,  1.22s/it]

 11%|█         | 125/1135 [05:45<20:36,  1.22s/it]

 11%|█         | 126/1135 [05:46<20:56,  1.25s/it]

 11%|█         | 127/1135 [05:47<20:30,  1.22s/it]

 11%|█▏        | 128/1135 [05:49<20:27,  1.22s/it]

 11%|█▏        | 129/1135 [05:50<20:13,  1.21s/it]

 11%|█▏        | 130/1135 [05:51<21:23,  1.28s/it]

 12%|█▏        | 131/1135 [05:52<21:18,  1.27s/it]

 12%|█▏        | 132/1135 [05:54<20:55,  1.25s/it]

 12%|█▏        | 133/1135 [05:55<20:48,  1.25s/it]

 12%|█▏        | 134/1135 [05:56<20:31,  1.23s/it]

 12%|█▏        | 135/1135 [05:57<20:17,  1.22s/it]

 12%|█▏        | 136/1135 [05:59<20:30,  1.23s/it]

 12%|█▏        | 137/1135 [06:00<20:23,  1.23s/it]

 12%|█▏        | 138/1135 [06:01<20:05,  1.21s/it]

 12%|█▏        | 139/1135 [06:02<19:51,  1.20s/it]

 12%|█▏        | 140/1135 [06:03<20:24,  1.23s/it]

 12%|█▏        | 141/1135 [06:05<20:08,  1.22s/it]

 13%|█▎        | 142/1135 [06:06<21:07,  1.28s/it]

 13%|█▎        | 143/1135 [06:07<22:00,  1.33s/it]

 13%|█▎        | 144/1135 [06:09<21:23,  1.30s/it]

 13%|█▎        | 145/1135 [06:10<20:47,  1.26s/it]

 13%|█▎        | 146/1135 [06:11<21:26,  1.30s/it]

 13%|█▎        | 147/1135 [06:13<22:53,  1.39s/it]

 13%|█▎        | 148/1135 [06:14<21:41,  1.32s/it]

 13%|█▎        | 149/1135 [06:15<20:49,  1.27s/it]

 13%|█▎        | 150/1135 [06:16<20:34,  1.25s/it]

 13%|█▎        | 151/1135 [06:18<20:15,  1.24s/it]

 13%|█▎        | 152/1135 [06:19<19:49,  1.21s/it]

 13%|█▎        | 153/1135 [06:20<19:35,  1.20s/it]

 14%|█▎        | 154/1135 [06:21<20:48,  1.27s/it]

 14%|█▎        | 155/1135 [06:23<21:07,  1.29s/it]

 14%|█▎        | 156/1135 [06:24<20:31,  1.26s/it]

 14%|█▍        | 157/1135 [06:25<20:17,  1.24s/it]

 14%|█▍        | 158/1135 [06:26<19:51,  1.22s/it]

 14%|█▍        | 159/1135 [06:27<20:12,  1.24s/it]

 14%|█▍        | 160/1135 [06:29<19:49,  1.22s/it]

 14%|█▍        | 161/1135 [06:30<19:39,  1.21s/it]

 14%|█▍        | 162/1135 [06:31<20:18,  1.25s/it]

 14%|█▍        | 163/1135 [06:32<19:45,  1.22s/it]

 14%|█▍        | 164/1135 [06:34<19:33,  1.21s/it]

 15%|█▍        | 165/1135 [06:35<20:12,  1.25s/it]

 15%|█▍        | 166/1135 [06:36<19:49,  1.23s/it]

 15%|█▍        | 167/1135 [06:37<20:03,  1.24s/it]

 15%|█▍        | 168/1135 [06:39<19:44,  1.22s/it]

 15%|█▍        | 169/1135 [06:40<19:26,  1.21s/it]

 15%|█▍        | 170/1135 [06:41<19:08,  1.19s/it]

 15%|█▌        | 171/1135 [06:42<19:12,  1.20s/it]

 15%|█▌        | 172/1135 [06:43<19:18,  1.20s/it]

 15%|█▌        | 173/1135 [06:44<19:29,  1.22s/it]

 15%|█▌        | 174/1135 [06:46<19:23,  1.21s/it]

 15%|█▌        | 175/1135 [06:47<20:17,  1.27s/it]

 16%|█▌        | 176/1135 [06:48<20:05,  1.26s/it]

 16%|█▌        | 177/1135 [06:49<19:39,  1.23s/it]

 16%|█▌        | 178/1135 [06:51<19:20,  1.21s/it]

 16%|█▌        | 179/1135 [06:52<18:58,  1.19s/it]

 16%|█▌        | 180/1135 [06:53<19:08,  1.20s/it]

 16%|█▌        | 181/1135 [06:54<18:57,  1.19s/it]

 16%|█▌        | 182/1135 [06:55<18:57,  1.19s/it]

 16%|█▌        | 183/1135 [06:57<19:45,  1.25s/it]

 16%|█▌        | 184/1135 [06:58<19:29,  1.23s/it]

 16%|█▋        | 185/1135 [06:59<19:11,  1.21s/it]

 16%|█▋        | 186/1135 [07:00<19:37,  1.24s/it]

 16%|█▋        | 187/1135 [07:02<20:50,  1.32s/it]

 17%|█▋        | 188/1135 [07:03<20:10,  1.28s/it]

 17%|█▋        | 189/1135 [07:04<19:35,  1.24s/it]

 17%|█▋        | 190/1135 [07:05<19:17,  1.23s/it]

 17%|█▋        | 191/1135 [07:07<18:50,  1.20s/it]

 17%|█▋        | 192/1135 [07:08<19:12,  1.22s/it]

 17%|█▋        | 193/1135 [07:09<19:03,  1.21s/it]

 17%|█▋        | 194/1135 [07:10<19:58,  1.27s/it]

 17%|█▋        | 195/1135 [07:12<19:21,  1.24s/it]

 17%|█▋        | 196/1135 [07:13<20:11,  1.29s/it]

 17%|█▋        | 197/1135 [07:15<21:15,  1.36s/it]

 17%|█▋        | 198/1135 [07:16<20:33,  1.32s/it]

 18%|█▊        | 199/1135 [07:17<19:54,  1.28s/it]

 18%|█▊        | 200/1135 [07:18<19:38,  1.26s/it]

 18%|█▊        | 201/1135 [07:20<22:21,  1.44s/it]

 18%|█▊        | 202/1135 [07:21<21:09,  1.36s/it]

 18%|█▊        | 203/1135 [07:22<20:21,  1.31s/it]

 18%|█▊        | 204/1135 [07:24<19:40,  1.27s/it]

 18%|█▊        | 205/1135 [07:25<19:47,  1.28s/it]

 18%|█▊        | 206/1135 [07:26<19:17,  1.25s/it]

 18%|█▊        | 207/1135 [07:27<19:03,  1.23s/it]

 18%|█▊        | 208/1135 [07:28<18:48,  1.22s/it]

 18%|█▊        | 209/1135 [07:30<18:30,  1.20s/it]

 19%|█▊        | 210/1135 [07:31<18:23,  1.19s/it]

 19%|█▊        | 211/1135 [07:32<18:25,  1.20s/it]

 19%|█▊        | 212/1135 [07:33<18:16,  1.19s/it]

 19%|█▉        | 213/1135 [07:34<18:15,  1.19s/it]

 19%|█▉        | 214/1135 [07:36<18:07,  1.18s/it]

 19%|█▉        | 215/1135 [07:37<18:21,  1.20s/it]

 19%|█▉        | 216/1135 [07:38<18:23,  1.20s/it]

 19%|█▉        | 217/1135 [07:39<18:40,  1.22s/it]

 19%|█▉        | 218/1135 [07:40<18:27,  1.21s/it]

 19%|█▉        | 219/1135 [07:42<18:12,  1.19s/it]

 19%|█▉        | 220/1135 [07:43<18:41,  1.23s/it]

 19%|█▉        | 221/1135 [07:44<18:24,  1.21s/it]

 20%|█▉        | 222/1135 [07:45<18:16,  1.20s/it]

 20%|█▉        | 223/1135 [07:47<18:48,  1.24s/it]

 20%|█▉        | 224/1135 [07:48<19:40,  1.30s/it]

 20%|█▉        | 225/1135 [07:49<19:07,  1.26s/it]

 20%|█▉        | 226/1135 [07:50<18:39,  1.23s/it]

 20%|██        | 227/1135 [07:52<18:45,  1.24s/it]

 20%|██        | 228/1135 [07:53<18:46,  1.24s/it]

 20%|██        | 229/1135 [07:54<18:27,  1.22s/it]

 20%|██        | 230/1135 [07:55<18:08,  1.20s/it]

 20%|██        | 231/1135 [07:56<17:58,  1.19s/it]

 20%|██        | 232/1135 [07:58<18:02,  1.20s/it]

 21%|██        | 233/1135 [07:59<17:57,  1.19s/it]

 21%|██        | 234/1135 [08:00<17:45,  1.18s/it]

 21%|██        | 235/1135 [08:01<17:48,  1.19s/it]

 21%|██        | 236/1135 [08:02<17:44,  1.18s/it]

 21%|██        | 237/1135 [08:03<17:51,  1.19s/it]

 21%|██        | 238/1135 [08:05<17:43,  1.19s/it]

 21%|██        | 239/1135 [08:06<17:40,  1.18s/it]

 21%|██        | 240/1135 [08:07<17:40,  1.18s/it]

 21%|██        | 241/1135 [08:08<17:33,  1.18s/it]

 21%|██▏       | 242/1135 [08:09<17:40,  1.19s/it]

 21%|██▏       | 243/1135 [08:11<17:44,  1.19s/it]

 21%|██▏       | 244/1135 [08:12<18:04,  1.22s/it]

 22%|██▏       | 245/1135 [08:13<19:47,  1.33s/it]

 22%|██▏       | 246/1135 [08:15<18:56,  1.28s/it]

 22%|██▏       | 247/1135 [08:16<18:21,  1.24s/it]

 22%|██▏       | 248/1135 [08:17<18:05,  1.22s/it]

 22%|██▏       | 249/1135 [08:18<17:40,  1.20s/it]

 22%|██▏       | 250/1135 [08:19<17:31,  1.19s/it]

 22%|██▏       | 251/1135 [08:20<17:42,  1.20s/it]

 22%|██▏       | 252/1135 [08:22<17:45,  1.21s/it]

 22%|██▏       | 253/1135 [08:23<17:30,  1.19s/it]

 22%|██▏       | 254/1135 [08:24<17:19,  1.18s/it]

 22%|██▏       | 255/1135 [08:25<17:11,  1.17s/it]

 23%|██▎       | 256/1135 [08:26<17:38,  1.20s/it]

 23%|██▎       | 257/1135 [08:28<17:19,  1.18s/it]

 23%|██▎       | 258/1135 [08:29<17:06,  1.17s/it]

 23%|██▎       | 259/1135 [08:30<17:17,  1.18s/it]

 23%|██▎       | 260/1135 [08:31<17:10,  1.18s/it]

 23%|██▎       | 261/1135 [08:32<17:00,  1.17s/it]

 23%|██▎       | 262/1135 [08:34<17:51,  1.23s/it]

 23%|██▎       | 263/1135 [08:35<17:53,  1.23s/it]

 23%|██▎       | 264/1135 [08:36<17:33,  1.21s/it]

 23%|██▎       | 265/1135 [08:37<17:16,  1.19s/it]

 23%|██▎       | 266/1135 [08:38<17:03,  1.18s/it]

 24%|██▎       | 267/1135 [08:39<17:03,  1.18s/it]

 24%|██▎       | 268/1135 [08:41<17:15,  1.19s/it]

 24%|██▎       | 269/1135 [08:42<16:58,  1.18s/it]

 24%|██▍       | 270/1135 [08:43<16:50,  1.17s/it]

 24%|██▍       | 271/1135 [08:44<17:30,  1.22s/it]

 24%|██▍       | 272/1135 [08:46<17:28,  1.22s/it]

 24%|██▍       | 273/1135 [08:47<18:18,  1.27s/it]

 24%|██▍       | 274/1135 [08:48<17:56,  1.25s/it]

 24%|██▍       | 275/1135 [08:49<17:37,  1.23s/it]

 24%|██▍       | 276/1135 [08:50<17:21,  1.21s/it]

 24%|██▍       | 277/1135 [08:52<17:16,  1.21s/it]

 24%|██▍       | 278/1135 [08:53<17:01,  1.19s/it]

 25%|██▍       | 279/1135 [08:54<16:44,  1.17s/it]

 25%|██▍       | 280/1135 [08:55<16:33,  1.16s/it]

 25%|██▍       | 281/1135 [08:56<16:52,  1.19s/it]

 25%|██▍       | 282/1135 [08:58<16:47,  1.18s/it]

 25%|██▍       | 283/1135 [08:59<16:42,  1.18s/it]

 25%|██▌       | 284/1135 [09:00<16:54,  1.19s/it]

 25%|██▌       | 285/1135 [09:01<16:49,  1.19s/it]

 25%|██▌       | 286/1135 [09:02<16:54,  1.19s/it]

 25%|██▌       | 287/1135 [09:04<17:25,  1.23s/it]

 25%|██▌       | 288/1135 [09:05<17:15,  1.22s/it]

 25%|██▌       | 289/1135 [09:06<16:59,  1.21s/it]

 26%|██▌       | 290/1135 [09:07<16:46,  1.19s/it]

 26%|██▌       | 291/1135 [09:08<16:56,  1.20s/it]

 26%|██▌       | 292/1135 [09:10<16:47,  1.19s/it]

 26%|██▌       | 293/1135 [09:11<16:32,  1.18s/it]

 26%|██▌       | 294/1135 [09:12<16:19,  1.16s/it]

 26%|██▌       | 295/1135 [09:13<16:17,  1.16s/it]

 26%|██▌       | 296/1135 [09:14<16:23,  1.17s/it]

 26%|██▌       | 297/1135 [09:15<16:17,  1.17s/it]

 26%|██▋       | 298/1135 [09:17<16:26,  1.18s/it]

 26%|██▋       | 299/1135 [09:18<16:32,  1.19s/it]

 26%|██▋       | 300/1135 [09:19<16:32,  1.19s/it]

 27%|██▋       | 301/1135 [09:20<16:16,  1.17s/it]

 27%|██▋       | 302/1135 [09:21<16:25,  1.18s/it]

 27%|██▋       | 303/1135 [09:22<16:27,  1.19s/it]

 27%|██▋       | 304/1135 [09:24<16:29,  1.19s/it]

 27%|██▋       | 305/1135 [09:25<16:18,  1.18s/it]

 27%|██▋       | 306/1135 [09:26<16:09,  1.17s/it]

 27%|██▋       | 307/1135 [09:27<16:18,  1.18s/it]

 27%|██▋       | 308/1135 [09:28<16:11,  1.18s/it]

 27%|██▋       | 309/1135 [09:30<17:22,  1.26s/it]

 27%|██▋       | 310/1135 [09:31<17:07,  1.25s/it]

 27%|██▋       | 311/1135 [09:32<16:44,  1.22s/it]

 27%|██▋       | 312/1135 [09:33<17:03,  1.24s/it]

 28%|██▊       | 313/1135 [09:35<16:58,  1.24s/it]

 28%|██▊       | 314/1135 [09:36<16:35,  1.21s/it]

 28%|██▊       | 315/1135 [09:37<16:18,  1.19s/it]

 28%|██▊       | 316/1135 [09:38<16:17,  1.19s/it]

 28%|██▊       | 317/1135 [09:39<16:29,  1.21s/it]

 28%|██▊       | 318/1135 [09:41<16:35,  1.22s/it]

 28%|██▊       | 319/1135 [09:42<16:26,  1.21s/it]

 28%|██▊       | 320/1135 [09:43<16:15,  1.20s/it]

 28%|██▊       | 321/1135 [09:44<16:03,  1.18s/it]

 28%|██▊       | 322/1135 [09:45<15:54,  1.17s/it]

 28%|██▊       | 323/1135 [09:47<17:44,  1.31s/it]

 29%|██▊       | 324/1135 [09:48<17:27,  1.29s/it]

 29%|██▊       | 325/1135 [09:50<18:53,  1.40s/it]

 29%|██▊       | 326/1135 [09:51<17:47,  1.32s/it]

 29%|██▉       | 327/1135 [09:52<17:09,  1.27s/it]

 29%|██▉       | 328/1135 [09:53<17:01,  1.27s/it]

 29%|██▉       | 329/1135 [09:55<16:34,  1.23s/it]

 29%|██▉       | 330/1135 [09:56<16:20,  1.22s/it]

 29%|██▉       | 331/1135 [09:57<17:04,  1.27s/it]

 29%|██▉       | 332/1135 [09:58<16:37,  1.24s/it]

 29%|██▉       | 333/1135 [09:59<16:14,  1.21s/it]

 29%|██▉       | 334/1135 [10:01<16:49,  1.26s/it]

 30%|██▉       | 335/1135 [10:02<16:41,  1.25s/it]

 30%|██▉       | 336/1135 [10:03<16:27,  1.24s/it]

 30%|██▉       | 337/1135 [10:05<19:18,  1.45s/it]

 30%|██▉       | 338/1135 [10:07<18:34,  1.40s/it]

 30%|██▉       | 339/1135 [10:08<17:51,  1.35s/it]

 30%|██▉       | 340/1135 [10:09<17:21,  1.31s/it]

 30%|███       | 341/1135 [10:10<16:58,  1.28s/it]

 30%|███       | 342/1135 [10:11<17:00,  1.29s/it]

 30%|███       | 343/1135 [10:13<18:37,  1.41s/it]

 30%|███       | 344/1135 [10:14<18:13,  1.38s/it]

 30%|███       | 345/1135 [10:17<21:30,  1.63s/it]

 30%|███       | 346/1135 [10:19<22:35,  1.72s/it]

 31%|███       | 347/1135 [10:20<20:56,  1.59s/it]

 31%|███       | 348/1135 [10:22<23:44,  1.81s/it]

 31%|███       | 349/1135 [10:23<21:19,  1.63s/it]

 31%|███       | 350/1135 [10:25<19:54,  1.52s/it]

 31%|███       | 351/1135 [10:26<18:38,  1.43s/it]

 31%|███       | 352/1135 [10:27<17:51,  1.37s/it]

 31%|███       | 353/1135 [10:28<17:10,  1.32s/it]

 31%|███       | 354/1135 [10:30<16:59,  1.31s/it]

 31%|███▏      | 355/1135 [10:32<19:57,  1.53s/it]

 31%|███▏      | 356/1135 [10:34<21:37,  1.67s/it]

 31%|███▏      | 357/1135 [10:36<25:52,  2.00s/it]

 32%|███▏      | 358/1135 [10:38<22:50,  1.76s/it]

 32%|███▏      | 359/1135 [10:39<21:24,  1.65s/it]

 32%|███▏      | 360/1135 [10:40<19:42,  1.53s/it]

 32%|███▏      | 361/1135 [10:43<23:02,  1.79s/it]

 32%|███▏      | 362/1135 [10:44<22:42,  1.76s/it]

 32%|███▏      | 363/1135 [10:46<20:32,  1.60s/it]

 32%|███▏      | 364/1135 [10:47<19:43,  1.53s/it]

 32%|███▏      | 365/1135 [10:48<18:34,  1.45s/it]

 32%|███▏      | 366/1135 [10:50<18:15,  1.43s/it]

 32%|███▏      | 367/1135 [10:51<17:37,  1.38s/it]

 32%|███▏      | 368/1135 [10:53<18:40,  1.46s/it]

 33%|███▎      | 369/1135 [10:54<19:33,  1.53s/it]

 33%|███▎      | 370/1135 [10:56<18:42,  1.47s/it]

 33%|███▎      | 371/1135 [10:57<18:01,  1.42s/it]

 33%|███▎      | 372/1135 [11:00<23:15,  1.83s/it]

 33%|███▎      | 373/1135 [11:01<20:54,  1.65s/it]

 33%|███▎      | 374/1135 [11:03<22:07,  1.75s/it]

 33%|███▎      | 375/1135 [11:05<22:43,  1.79s/it]

 33%|███▎      | 376/1135 [11:06<20:50,  1.65s/it]

 33%|███▎      | 377/1135 [11:07<19:28,  1.54s/it]

 33%|███▎      | 378/1135 [11:09<18:15,  1.45s/it]

 33%|███▎      | 379/1135 [11:10<17:21,  1.38s/it]

 33%|███▎      | 380/1135 [11:11<16:48,  1.34s/it]

 34%|███▎      | 381/1135 [11:12<16:17,  1.30s/it]

 34%|███▎      | 382/1135 [11:13<16:05,  1.28s/it]

 34%|███▎      | 383/1135 [11:15<15:56,  1.27s/it]

 34%|███▍      | 384/1135 [11:16<15:40,  1.25s/it]

 34%|███▍      | 385/1135 [11:17<15:38,  1.25s/it]

 34%|███▍      | 386/1135 [11:19<19:23,  1.55s/it]

 34%|███▍      | 387/1135 [11:21<20:37,  1.65s/it]

 34%|███▍      | 388/1135 [11:23<18:59,  1.53s/it]

 34%|███▍      | 389/1135 [11:24<17:46,  1.43s/it]

 34%|███▍      | 390/1135 [11:25<16:53,  1.36s/it]

 34%|███▍      | 391/1135 [11:26<16:31,  1.33s/it]

 35%|███▍      | 392/1135 [11:27<16:07,  1.30s/it]

 35%|███▍      | 393/1135 [11:29<15:33,  1.26s/it]

 35%|███▍      | 394/1135 [11:30<15:37,  1.26s/it]

 35%|███▍      | 395/1135 [11:31<15:24,  1.25s/it]

 35%|███▍      | 396/1135 [11:32<15:09,  1.23s/it]

 35%|███▍      | 397/1135 [11:34<15:13,  1.24s/it]

 35%|███▌      | 398/1135 [11:35<15:03,  1.23s/it]

 35%|███▌      | 399/1135 [11:36<14:55,  1.22s/it]

 35%|███▌      | 400/1135 [11:37<15:14,  1.24s/it]

 35%|███▌      | 401/1135 [11:38<15:05,  1.23s/it]

 35%|███▌      | 402/1135 [11:40<14:53,  1.22s/it]

 36%|███▌      | 403/1135 [11:41<15:10,  1.24s/it]

 36%|███▌      | 404/1135 [11:42<15:08,  1.24s/it]

 36%|███▌      | 405/1135 [11:44<15:48,  1.30s/it]

 36%|███▌      | 406/1135 [11:45<16:26,  1.35s/it]

 36%|███▌      | 407/1135 [11:46<16:07,  1.33s/it]

 36%|███▌      | 408/1135 [11:48<15:44,  1.30s/it]

 36%|███▌      | 409/1135 [11:49<15:14,  1.26s/it]

 36%|███▌      | 410/1135 [11:50<14:59,  1.24s/it]

 36%|███▌      | 411/1135 [11:51<14:48,  1.23s/it]

 36%|███▋      | 412/1135 [11:52<15:13,  1.26s/it]

 36%|███▋      | 413/1135 [11:54<15:00,  1.25s/it]

 36%|███▋      | 414/1135 [11:55<14:47,  1.23s/it]

 37%|███▋      | 415/1135 [11:56<15:14,  1.27s/it]

 37%|███▋      | 416/1135 [11:57<15:01,  1.25s/it]

 37%|███▋      | 417/1135 [11:59<14:59,  1.25s/it]

 37%|███▋      | 418/1135 [12:00<15:04,  1.26s/it]

 37%|███▋      | 419/1135 [12:02<16:05,  1.35s/it]

 37%|███▋      | 420/1135 [12:03<15:29,  1.30s/it]

 37%|███▋      | 421/1135 [12:04<15:26,  1.30s/it]

 37%|███▋      | 422/1135 [12:05<15:04,  1.27s/it]

 37%|███▋      | 423/1135 [12:06<14:57,  1.26s/it]

 37%|███▋      | 424/1135 [12:08<15:38,  1.32s/it]

 37%|███▋      | 425/1135 [12:09<15:04,  1.27s/it]

 38%|███▊      | 426/1135 [12:11<15:57,  1.35s/it]

 38%|███▊      | 427/1135 [12:12<16:31,  1.40s/it]

 38%|███▊      | 428/1135 [12:13<15:39,  1.33s/it]

 38%|███▊      | 429/1135 [12:16<19:44,  1.68s/it]

 38%|███▊      | 430/1135 [12:17<18:07,  1.54s/it]

 38%|███▊      | 431/1135 [12:18<16:47,  1.43s/it]

 38%|███▊      | 432/1135 [12:19<16:04,  1.37s/it]

 38%|███▊      | 433/1135 [12:21<15:20,  1.31s/it]

 38%|███▊      | 434/1135 [12:22<14:57,  1.28s/it]

 38%|███▊      | 435/1135 [12:23<14:49,  1.27s/it]

 38%|███▊      | 436/1135 [12:24<14:42,  1.26s/it]

 39%|███▊      | 437/1135 [12:26<14:35,  1.25s/it]

 39%|███▊      | 438/1135 [12:27<14:24,  1.24s/it]

 39%|███▊      | 439/1135 [12:28<14:12,  1.22s/it]

 39%|███▉      | 440/1135 [12:29<14:20,  1.24s/it]

 39%|███▉      | 441/1135 [12:30<14:26,  1.25s/it]

 39%|███▉      | 442/1135 [12:32<14:14,  1.23s/it]

 39%|███▉      | 443/1135 [12:33<13:57,  1.21s/it]

 39%|███▉      | 444/1135 [12:34<14:04,  1.22s/it]

 39%|███▉      | 445/1135 [12:35<13:57,  1.21s/it]

 39%|███▉      | 446/1135 [12:37<13:59,  1.22s/it]

 39%|███▉      | 447/1135 [12:38<14:39,  1.28s/it]

 39%|███▉      | 448/1135 [12:39<14:19,  1.25s/it]

 40%|███▉      | 449/1135 [12:40<14:18,  1.25s/it]

 40%|███▉      | 450/1135 [12:42<15:27,  1.35s/it]

 40%|███▉      | 451/1135 [12:43<14:59,  1.31s/it]

 40%|███▉      | 452/1135 [12:44<14:40,  1.29s/it]

 40%|███▉      | 453/1135 [12:46<14:15,  1.25s/it]

 40%|████      | 454/1135 [12:47<14:10,  1.25s/it]

 40%|████      | 455/1135 [12:48<15:18,  1.35s/it]

 40%|████      | 456/1135 [12:50<14:45,  1.30s/it]

 40%|████      | 457/1135 [12:51<14:34,  1.29s/it]

 40%|████      | 458/1135 [12:52<15:28,  1.37s/it]

 40%|████      | 459/1135 [12:54<14:55,  1.32s/it]

 41%|████      | 460/1135 [12:55<14:32,  1.29s/it]

 41%|████      | 461/1135 [12:56<14:18,  1.27s/it]

 41%|████      | 462/1135 [12:58<15:17,  1.36s/it]

 41%|████      | 463/1135 [12:59<14:44,  1.32s/it]

 41%|████      | 464/1135 [13:00<14:39,  1.31s/it]

 41%|████      | 465/1135 [13:01<14:41,  1.32s/it]

 41%|████      | 466/1135 [13:03<14:15,  1.28s/it]

 41%|████      | 467/1135 [13:04<14:38,  1.32s/it]

 41%|████      | 468/1135 [13:05<14:03,  1.26s/it]

 41%|████▏     | 469/1135 [13:06<13:53,  1.25s/it]

 41%|████▏     | 470/1135 [13:08<13:38,  1.23s/it]

 41%|████▏     | 471/1135 [13:09<13:25,  1.21s/it]

 42%|████▏     | 472/1135 [13:10<13:35,  1.23s/it]

 42%|████▏     | 473/1135 [13:11<13:35,  1.23s/it]

 42%|████▏     | 474/1135 [13:14<17:34,  1.59s/it]

 42%|████▏     | 475/1135 [13:15<16:19,  1.48s/it]

 42%|████▏     | 476/1135 [13:16<15:17,  1.39s/it]

 42%|████▏     | 477/1135 [13:17<14:53,  1.36s/it]

 42%|████▏     | 478/1135 [13:19<15:06,  1.38s/it]

 42%|████▏     | 479/1135 [13:20<14:22,  1.32s/it]

 42%|████▏     | 480/1135 [13:21<14:09,  1.30s/it]

 42%|████▏     | 481/1135 [13:22<13:46,  1.26s/it]

 42%|████▏     | 482/1135 [13:24<13:42,  1.26s/it]

 43%|████▎     | 483/1135 [13:25<13:39,  1.26s/it]

 43%|████▎     | 484/1135 [13:26<13:27,  1.24s/it]

 43%|████▎     | 485/1135 [13:27<13:31,  1.25s/it]

 43%|████▎     | 486/1135 [13:29<13:15,  1.23s/it]

 43%|████▎     | 487/1135 [13:30<13:04,  1.21s/it]

 43%|████▎     | 488/1135 [13:31<13:07,  1.22s/it]

 43%|████▎     | 489/1135 [13:32<13:12,  1.23s/it]

 43%|████▎     | 490/1135 [13:33<13:09,  1.22s/it]

 43%|████▎     | 491/1135 [13:35<13:07,  1.22s/it]

 43%|████▎     | 492/1135 [13:36<13:24,  1.25s/it]

 43%|████▎     | 493/1135 [13:37<13:05,  1.22s/it]

 44%|████▎     | 494/1135 [13:38<13:10,  1.23s/it]

 44%|████▎     | 495/1135 [13:40<12:58,  1.22s/it]

 44%|████▎     | 496/1135 [13:41<12:50,  1.21s/it]

 44%|████▍     | 497/1135 [13:42<12:44,  1.20s/it]

 44%|████▍     | 498/1135 [13:43<12:37,  1.19s/it]

 44%|████▍     | 499/1135 [13:44<12:36,  1.19s/it]

 44%|████▍     | 500/1135 [13:46<12:35,  1.19s/it]

 44%|████▍     | 501/1135 [13:47<12:32,  1.19s/it]

 44%|████▍     | 502/1135 [13:48<12:28,  1.18s/it]

 44%|████▍     | 503/1135 [13:49<12:28,  1.18s/it]

 44%|████▍     | 504/1135 [13:50<12:26,  1.18s/it]

 44%|████▍     | 505/1135 [13:51<12:33,  1.20s/it]

 45%|████▍     | 506/1135 [13:53<12:42,  1.21s/it]

 45%|████▍     | 507/1135 [13:54<12:40,  1.21s/it]

 45%|████▍     | 508/1135 [13:55<13:43,  1.31s/it]

 45%|████▍     | 509/1135 [13:57<13:19,  1.28s/it]

 45%|████▍     | 510/1135 [13:58<13:29,  1.30s/it]

 45%|████▌     | 511/1135 [13:59<13:03,  1.26s/it]

 45%|████▌     | 512/1135 [14:00<12:59,  1.25s/it]

 45%|████▌     | 513/1135 [14:02<12:55,  1.25s/it]

 45%|████▌     | 514/1135 [14:03<12:45,  1.23s/it]

 45%|████▌     | 515/1135 [14:04<12:36,  1.22s/it]

 45%|████▌     | 516/1135 [14:05<12:42,  1.23s/it]

 46%|████▌     | 517/1135 [14:07<12:38,  1.23s/it]

 46%|████▌     | 518/1135 [14:08<12:25,  1.21s/it]

 46%|████▌     | 519/1135 [14:09<12:21,  1.20s/it]

 46%|████▌     | 520/1135 [14:10<12:32,  1.22s/it]

 46%|████▌     | 521/1135 [14:11<12:24,  1.21s/it]

 46%|████▌     | 522/1135 [14:13<12:40,  1.24s/it]

 46%|████▌     | 523/1135 [14:14<12:30,  1.23s/it]

 46%|████▌     | 524/1135 [14:15<12:23,  1.22s/it]

 46%|████▋     | 525/1135 [14:16<13:05,  1.29s/it]

 46%|████▋     | 526/1135 [14:18<13:01,  1.28s/it]

 46%|████▋     | 527/1135 [14:19<12:42,  1.25s/it]

 47%|████▋     | 528/1135 [14:20<12:33,  1.24s/it]

 47%|████▋     | 529/1135 [14:21<12:22,  1.23s/it]

 47%|████▋     | 530/1135 [14:23<12:50,  1.27s/it]

 47%|████▋     | 531/1135 [14:24<12:33,  1.25s/it]

 47%|████▋     | 532/1135 [14:25<12:54,  1.28s/it]

 47%|████▋     | 533/1135 [14:26<12:38,  1.26s/it]

 47%|████▋     | 534/1135 [14:28<12:33,  1.25s/it]

 47%|████▋     | 535/1135 [14:29<12:24,  1.24s/it]

 47%|████▋     | 536/1135 [14:30<12:09,  1.22s/it]

 47%|████▋     | 537/1135 [14:31<12:02,  1.21s/it]

 47%|████▋     | 538/1135 [14:33<13:30,  1.36s/it]

 47%|████▋     | 539/1135 [14:34<12:58,  1.31s/it]

 48%|████▊     | 540/1135 [14:35<12:36,  1.27s/it]

 48%|████▊     | 541/1135 [14:37<12:17,  1.24s/it]

 48%|████▊     | 542/1135 [14:38<12:19,  1.25s/it]

 48%|████▊     | 543/1135 [14:39<12:10,  1.23s/it]

 48%|████▊     | 544/1135 [14:40<11:58,  1.21s/it]

 48%|████▊     | 545/1135 [14:41<11:49,  1.20s/it]

 48%|████▊     | 546/1135 [14:43<11:43,  1.19s/it]

 48%|████▊     | 547/1135 [14:44<11:38,  1.19s/it]

 48%|████▊     | 548/1135 [14:45<11:40,  1.19s/it]

 48%|████▊     | 549/1135 [14:46<11:58,  1.23s/it]

 48%|████▊     | 550/1135 [14:47<12:00,  1.23s/it]

 49%|████▊     | 551/1135 [14:49<11:45,  1.21s/it]

 49%|████▊     | 552/1135 [14:50<11:55,  1.23s/it]

 49%|████▊     | 553/1135 [14:51<11:59,  1.24s/it]

 49%|████▉     | 554/1135 [14:52<11:45,  1.22s/it]

 49%|████▉     | 555/1135 [14:54<11:49,  1.22s/it]

 49%|████▉     | 556/1135 [14:55<12:10,  1.26s/it]

 49%|████▉     | 557/1135 [14:56<12:09,  1.26s/it]

 49%|████▉     | 558/1135 [14:57<11:58,  1.25s/it]

 49%|████▉     | 559/1135 [14:59<11:46,  1.23s/it]

 49%|████▉     | 560/1135 [15:00<11:55,  1.24s/it]

 49%|████▉     | 561/1135 [15:01<11:44,  1.23s/it]

 50%|████▉     | 562/1135 [15:02<11:32,  1.21s/it]

 50%|████▉     | 563/1135 [15:03<11:49,  1.24s/it]

 50%|████▉     | 564/1135 [15:05<11:48,  1.24s/it]

 50%|████▉     | 565/1135 [15:06<11:39,  1.23s/it]

 50%|████▉     | 566/1135 [15:07<11:45,  1.24s/it]

 50%|████▉     | 567/1135 [15:08<11:32,  1.22s/it]

 50%|█████     | 568/1135 [15:10<12:19,  1.30s/it]

 50%|█████     | 569/1135 [15:11<12:06,  1.28s/it]

 50%|█████     | 570/1135 [15:12<11:45,  1.25s/it]

 50%|█████     | 571/1135 [15:14<11:41,  1.24s/it]

 50%|█████     | 572/1135 [15:15<11:36,  1.24s/it]

 50%|█████     | 573/1135 [15:16<11:23,  1.22s/it]

 51%|█████     | 574/1135 [15:17<11:23,  1.22s/it]

 51%|█████     | 575/1135 [15:18<11:26,  1.23s/it]

 51%|█████     | 576/1135 [15:20<11:19,  1.22s/it]

 51%|█████     | 577/1135 [15:21<11:22,  1.22s/it]

 51%|█████     | 578/1135 [15:22<11:17,  1.22s/it]

 51%|█████     | 579/1135 [15:23<11:08,  1.20s/it]

 51%|█████     | 580/1135 [15:24<11:05,  1.20s/it]

 51%|█████     | 581/1135 [15:26<11:02,  1.20s/it]

 51%|█████▏    | 582/1135 [15:27<11:04,  1.20s/it]

 51%|█████▏    | 583/1135 [15:28<11:09,  1.21s/it]

 51%|█████▏    | 584/1135 [15:29<11:05,  1.21s/it]

 52%|█████▏    | 585/1135 [15:30<11:11,  1.22s/it]

 52%|█████▏    | 586/1135 [15:32<11:10,  1.22s/it]

 52%|█████▏    | 587/1135 [15:33<11:07,  1.22s/it]

 52%|█████▏    | 588/1135 [15:34<11:01,  1.21s/it]

 52%|█████▏    | 589/1135 [15:35<10:54,  1.20s/it]

 52%|█████▏    | 590/1135 [15:37<11:09,  1.23s/it]

 52%|█████▏    | 591/1135 [15:38<10:57,  1.21s/it]

 52%|█████▏    | 592/1135 [15:39<10:57,  1.21s/it]

 52%|█████▏    | 593/1135 [15:40<11:31,  1.28s/it]

 52%|█████▏    | 594/1135 [15:42<11:20,  1.26s/it]

 52%|█████▏    | 595/1135 [15:43<11:18,  1.26s/it]

 53%|█████▎    | 596/1135 [15:44<11:10,  1.24s/it]

 53%|█████▎    | 597/1135 [15:45<10:59,  1.23s/it]

 53%|█████▎    | 598/1135 [15:46<11:04,  1.24s/it]

 53%|█████▎    | 599/1135 [15:48<10:57,  1.23s/it]

 53%|█████▎    | 600/1135 [15:49<10:48,  1.21s/it]

 53%|█████▎    | 601/1135 [15:50<10:40,  1.20s/it]

 53%|█████▎    | 602/1135 [15:51<10:40,  1.20s/it]

 53%|█████▎    | 603/1135 [15:52<10:44,  1.21s/it]

 53%|█████▎    | 604/1135 [15:54<10:41,  1.21s/it]

 53%|█████▎    | 605/1135 [15:55<10:45,  1.22s/it]

 53%|█████▎    | 606/1135 [15:56<10:46,  1.22s/it]

 53%|█████▎    | 607/1135 [15:57<10:43,  1.22s/it]

 54%|█████▎    | 608/1135 [15:59<10:42,  1.22s/it]

 54%|█████▎    | 609/1135 [16:00<10:40,  1.22s/it]

 54%|█████▎    | 610/1135 [16:01<10:51,  1.24s/it]

 54%|█████▍    | 611/1135 [16:02<11:09,  1.28s/it]

 54%|█████▍    | 612/1135 [16:04<11:02,  1.27s/it]

 54%|█████▍    | 613/1135 [16:05<11:05,  1.28s/it]

 54%|█████▍    | 614/1135 [16:06<10:50,  1.25s/it]

 54%|█████▍    | 615/1135 [16:07<10:53,  1.26s/it]

 54%|█████▍    | 616/1135 [16:09<10:37,  1.23s/it]

 54%|█████▍    | 617/1135 [16:10<10:31,  1.22s/it]

 54%|█████▍    | 618/1135 [16:11<10:25,  1.21s/it]

 55%|█████▍    | 619/1135 [16:12<10:27,  1.22s/it]

 55%|█████▍    | 620/1135 [16:13<10:28,  1.22s/it]

 55%|█████▍    | 621/1135 [16:15<10:48,  1.26s/it]

 55%|█████▍    | 622/1135 [16:16<11:11,  1.31s/it]

 55%|█████▍    | 623/1135 [16:17<10:56,  1.28s/it]

 55%|█████▍    | 624/1135 [16:19<10:43,  1.26s/it]

 55%|█████▌    | 625/1135 [16:20<10:28,  1.23s/it]

 55%|█████▌    | 626/1135 [16:21<10:33,  1.25s/it]

 55%|█████▌    | 627/1135 [16:22<10:25,  1.23s/it]

 55%|█████▌    | 628/1135 [16:23<10:15,  1.21s/it]

 55%|█████▌    | 629/1135 [16:25<10:24,  1.23s/it]

 56%|█████▌    | 630/1135 [16:26<11:35,  1.38s/it]

 56%|█████▌    | 631/1135 [16:28<11:07,  1.32s/it]

 56%|█████▌    | 632/1135 [16:29<11:08,  1.33s/it]

 56%|█████▌    | 633/1135 [16:30<10:47,  1.29s/it]

 56%|█████▌    | 634/1135 [16:31<10:36,  1.27s/it]

 56%|█████▌    | 635/1135 [16:33<10:24,  1.25s/it]

 56%|█████▌    | 636/1135 [16:34<10:13,  1.23s/it]

 56%|█████▌    | 637/1135 [16:35<10:15,  1.24s/it]

 56%|█████▌    | 638/1135 [16:37<10:57,  1.32s/it]

 56%|█████▋    | 639/1135 [16:38<11:00,  1.33s/it]

 56%|█████▋    | 640/1135 [16:39<10:57,  1.33s/it]

 56%|█████▋    | 641/1135 [16:41<10:42,  1.30s/it]

 57%|█████▋    | 642/1135 [16:42<10:33,  1.28s/it]

 57%|█████▋    | 643/1135 [16:43<10:18,  1.26s/it]

 57%|█████▋    | 644/1135 [16:44<10:11,  1.25s/it]

 57%|█████▋    | 645/1135 [16:45<10:12,  1.25s/it]

 57%|█████▋    | 646/1135 [16:47<12:11,  1.50s/it]

 57%|█████▋    | 647/1135 [16:49<11:37,  1.43s/it]

 57%|█████▋    | 648/1135 [16:50<11:30,  1.42s/it]

 57%|█████▋    | 649/1135 [16:51<11:01,  1.36s/it]

 57%|█████▋    | 650/1135 [16:53<12:08,  1.50s/it]

 57%|█████▋    | 651/1135 [16:57<18:36,  2.31s/it]

 57%|█████▋    | 652/1135 [17:00<20:22,  2.53s/it]

 58%|█████▊    | 653/1135 [17:03<20:29,  2.55s/it]

 58%|█████▊    | 654/1135 [17:04<17:12,  2.15s/it]

 58%|█████▊    | 655/1135 [17:06<15:30,  1.94s/it]

 58%|█████▊    | 656/1135 [17:07<14:45,  1.85s/it]

 58%|█████▊    | 657/1135 [17:09<13:31,  1.70s/it]

 58%|█████▊    | 658/1135 [17:10<12:52,  1.62s/it]

 58%|█████▊    | 659/1135 [17:12<12:54,  1.63s/it]

 58%|█████▊    | 660/1135 [17:14<13:16,  1.68s/it]

 58%|█████▊    | 661/1135 [17:15<12:05,  1.53s/it]

 58%|█████▊    | 662/1135 [17:16<11:20,  1.44s/it]

 58%|█████▊    | 663/1135 [17:17<10:45,  1.37s/it]

 59%|█████▊    | 664/1135 [17:18<10:18,  1.31s/it]

 59%|█████▊    | 665/1135 [17:20<09:57,  1.27s/it]

 59%|█████▊    | 666/1135 [17:21<10:07,  1.30s/it]

 59%|█████▉    | 667/1135 [17:22<09:53,  1.27s/it]

 59%|█████▉    | 668/1135 [17:23<09:38,  1.24s/it]

 59%|█████▉    | 669/1135 [17:25<09:39,  1.24s/it]

 59%|█████▉    | 670/1135 [17:26<09:31,  1.23s/it]

 59%|█████▉    | 671/1135 [17:27<09:23,  1.21s/it]

 59%|█████▉    | 672/1135 [17:28<09:16,  1.20s/it]

 59%|█████▉    | 673/1135 [17:29<09:13,  1.20s/it]

 59%|█████▉    | 674/1135 [17:30<09:13,  1.20s/it]

 59%|█████▉    | 675/1135 [17:32<09:10,  1.20s/it]

 60%|█████▉    | 676/1135 [17:33<09:25,  1.23s/it]

 60%|█████▉    | 677/1135 [17:34<09:16,  1.21s/it]

 60%|█████▉    | 678/1135 [17:35<09:22,  1.23s/it]

 60%|█████▉    | 679/1135 [17:37<09:13,  1.21s/it]

 60%|█████▉    | 680/1135 [17:38<09:16,  1.22s/it]

 60%|██████    | 681/1135 [17:39<09:12,  1.22s/it]

 60%|██████    | 682/1135 [17:40<09:11,  1.22s/it]

 60%|██████    | 683/1135 [17:41<09:06,  1.21s/it]

 60%|██████    | 684/1135 [17:43<08:57,  1.19s/it]

 60%|██████    | 685/1135 [17:44<08:55,  1.19s/it]

 60%|██████    | 686/1135 [17:45<08:59,  1.20s/it]

 61%|██████    | 687/1135 [17:46<08:57,  1.20s/it]

 61%|██████    | 688/1135 [17:47<08:52,  1.19s/it]

 61%|██████    | 689/1135 [17:49<09:01,  1.21s/it]

 61%|██████    | 690/1135 [17:50<08:56,  1.21s/it]

 61%|██████    | 691/1135 [17:51<08:49,  1.19s/it]

 61%|██████    | 692/1135 [17:52<09:04,  1.23s/it]

 61%|██████    | 693/1135 [17:54<09:00,  1.22s/it]

 61%|██████    | 694/1135 [17:55<08:54,  1.21s/it]

 61%|██████    | 695/1135 [17:56<08:48,  1.20s/it]

 61%|██████▏   | 696/1135 [17:57<08:40,  1.19s/it]

 61%|██████▏   | 697/1135 [17:58<08:59,  1.23s/it]

 61%|██████▏   | 698/1135 [18:00<08:49,  1.21s/it]

 62%|██████▏   | 699/1135 [18:01<08:49,  1.21s/it]

 62%|██████▏   | 700/1135 [18:02<08:41,  1.20s/it]

 62%|██████▏   | 701/1135 [18:03<08:38,  1.20s/it]

 62%|██████▏   | 702/1135 [18:04<08:47,  1.22s/it]

 62%|██████▏   | 703/1135 [18:06<08:42,  1.21s/it]

 62%|██████▏   | 704/1135 [18:07<08:35,  1.20s/it]

 62%|██████▏   | 705/1135 [18:08<08:29,  1.19s/it]

 62%|██████▏   | 706/1135 [18:09<08:30,  1.19s/it]

 62%|██████▏   | 707/1135 [18:10<08:30,  1.19s/it]

 62%|██████▏   | 708/1135 [18:12<08:31,  1.20s/it]

 62%|██████▏   | 709/1135 [18:13<08:37,  1.22s/it]

 63%|██████▎   | 710/1135 [18:14<08:31,  1.20s/it]

 63%|██████▎   | 711/1135 [18:15<08:27,  1.20s/it]

 63%|██████▎   | 712/1135 [18:16<08:21,  1.19s/it]

 63%|██████▎   | 713/1135 [18:18<08:26,  1.20s/it]

 63%|██████▎   | 714/1135 [18:19<08:32,  1.22s/it]

 63%|██████▎   | 715/1135 [18:20<08:31,  1.22s/it]

 63%|██████▎   | 716/1135 [18:21<09:01,  1.29s/it]

 63%|██████▎   | 717/1135 [18:23<09:11,  1.32s/it]

 63%|██████▎   | 718/1135 [18:24<08:52,  1.28s/it]

 63%|██████▎   | 719/1135 [18:25<08:40,  1.25s/it]

 63%|██████▎   | 720/1135 [18:26<08:34,  1.24s/it]

 64%|██████▎   | 721/1135 [18:28<08:23,  1.22s/it]

 64%|██████▎   | 722/1135 [18:29<08:18,  1.21s/it]

 64%|██████▎   | 723/1135 [18:30<08:11,  1.19s/it]

 64%|██████▍   | 724/1135 [18:31<08:11,  1.20s/it]

 64%|██████▍   | 725/1135 [18:32<08:08,  1.19s/it]

 64%|██████▍   | 726/1135 [18:33<08:07,  1.19s/it]

 64%|██████▍   | 727/1135 [18:35<08:08,  1.20s/it]

 64%|██████▍   | 728/1135 [18:36<08:09,  1.20s/it]

 64%|██████▍   | 729/1135 [18:37<08:03,  1.19s/it]

 64%|██████▍   | 730/1135 [18:38<07:57,  1.18s/it]

 64%|██████▍   | 731/1135 [18:40<08:08,  1.21s/it]

 64%|██████▍   | 732/1135 [18:41<08:08,  1.21s/it]

 65%|██████▍   | 733/1135 [18:42<08:01,  1.20s/it]

 65%|██████▍   | 734/1135 [18:43<08:09,  1.22s/it]

 65%|██████▍   | 735/1135 [18:44<08:03,  1.21s/it]

 65%|██████▍   | 736/1135 [18:46<07:57,  1.20s/it]

 65%|██████▍   | 737/1135 [18:47<07:53,  1.19s/it]

 65%|██████▌   | 738/1135 [18:48<07:54,  1.20s/it]

 65%|██████▌   | 739/1135 [18:49<07:50,  1.19s/it]

 65%|██████▌   | 740/1135 [18:50<07:46,  1.18s/it]

 65%|██████▌   | 741/1135 [18:51<07:42,  1.17s/it]

 65%|██████▌   | 742/1135 [18:53<07:47,  1.19s/it]

 65%|██████▌   | 743/1135 [18:54<07:43,  1.18s/it]

 66%|██████▌   | 744/1135 [18:55<07:41,  1.18s/it]

 66%|██████▌   | 745/1135 [18:56<07:39,  1.18s/it]

 66%|██████▌   | 746/1135 [18:57<07:40,  1.18s/it]

 66%|██████▌   | 747/1135 [18:59<07:47,  1.20s/it]

 66%|██████▌   | 748/1135 [19:00<07:40,  1.19s/it]

 66%|██████▌   | 749/1135 [19:01<07:48,  1.21s/it]

 66%|██████▌   | 750/1135 [19:02<07:42,  1.20s/it]

 66%|██████▌   | 751/1135 [19:03<07:43,  1.21s/it]

 66%|██████▋   | 752/1135 [19:05<07:56,  1.24s/it]

 66%|██████▋   | 753/1135 [19:06<07:46,  1.22s/it]

 66%|██████▋   | 754/1135 [19:07<07:41,  1.21s/it]

 67%|██████▋   | 755/1135 [19:08<07:45,  1.22s/it]

 67%|██████▋   | 756/1135 [19:10<07:38,  1.21s/it]

 67%|██████▋   | 757/1135 [19:11<07:32,  1.20s/it]

 67%|██████▋   | 758/1135 [19:12<07:29,  1.19s/it]

 67%|██████▋   | 759/1135 [19:13<07:29,  1.20s/it]

 67%|██████▋   | 760/1135 [19:14<07:29,  1.20s/it]

 67%|██████▋   | 761/1135 [19:15<07:28,  1.20s/it]

 67%|██████▋   | 762/1135 [19:17<07:41,  1.24s/it]

 67%|██████▋   | 763/1135 [19:18<07:38,  1.23s/it]

 67%|██████▋   | 764/1135 [19:19<07:38,  1.24s/it]

 67%|██████▋   | 765/1135 [19:21<07:37,  1.24s/it]

 67%|██████▋   | 766/1135 [19:22<07:40,  1.25s/it]

 68%|██████▊   | 767/1135 [19:23<07:37,  1.24s/it]

 68%|██████▊   | 768/1135 [19:24<07:27,  1.22s/it]

 68%|██████▊   | 769/1135 [19:25<07:28,  1.23s/it]

 68%|██████▊   | 770/1135 [19:27<07:18,  1.20s/it]

 68%|██████▊   | 771/1135 [19:28<07:14,  1.19s/it]

 68%|██████▊   | 772/1135 [19:29<07:11,  1.19s/it]

 68%|██████▊   | 773/1135 [19:30<07:09,  1.19s/it]

 68%|██████▊   | 774/1135 [19:31<07:04,  1.18s/it]

 68%|██████▊   | 775/1135 [19:32<07:08,  1.19s/it]

 68%|██████▊   | 776/1135 [19:34<07:07,  1.19s/it]

 68%|██████▊   | 777/1135 [19:35<07:07,  1.19s/it]

 69%|██████▊   | 778/1135 [19:36<07:14,  1.22s/it]

 69%|██████▊   | 779/1135 [19:37<07:21,  1.24s/it]

 69%|██████▊   | 780/1135 [19:39<07:15,  1.23s/it]

 69%|██████▉   | 781/1135 [19:40<07:17,  1.24s/it]

 69%|██████▉   | 782/1135 [19:41<07:18,  1.24s/it]

 69%|██████▉   | 783/1135 [19:42<07:08,  1.22s/it]

 69%|██████▉   | 784/1135 [19:43<07:03,  1.21s/it]

 69%|██████▉   | 785/1135 [19:45<07:05,  1.21s/it]

 69%|██████▉   | 786/1135 [19:46<07:00,  1.20s/it]

 69%|██████▉   | 787/1135 [19:47<07:19,  1.26s/it]

 69%|██████▉   | 788/1135 [19:49<07:13,  1.25s/it]

 70%|██████▉   | 789/1135 [19:50<07:05,  1.23s/it]

 70%|██████▉   | 790/1135 [19:51<07:07,  1.24s/it]

 70%|██████▉   | 791/1135 [19:52<07:07,  1.24s/it]

 70%|██████▉   | 792/1135 [19:53<07:03,  1.24s/it]

 70%|██████▉   | 793/1135 [19:55<07:14,  1.27s/it]

 70%|██████▉   | 794/1135 [19:56<07:46,  1.37s/it]

 70%|███████   | 795/1135 [19:58<07:32,  1.33s/it]

 70%|███████   | 796/1135 [19:59<07:20,  1.30s/it]

 70%|███████   | 797/1135 [20:00<07:06,  1.26s/it]

 70%|███████   | 798/1135 [20:01<07:09,  1.27s/it]

 70%|███████   | 799/1135 [20:03<07:02,  1.26s/it]

 70%|███████   | 800/1135 [20:04<07:15,  1.30s/it]

 71%|███████   | 801/1135 [20:05<07:03,  1.27s/it]

 71%|███████   | 802/1135 [20:06<07:02,  1.27s/it]

 71%|███████   | 803/1135 [20:08<06:52,  1.24s/it]

 71%|███████   | 804/1135 [20:09<06:45,  1.23s/it]

 71%|███████   | 805/1135 [20:10<07:04,  1.29s/it]

 71%|███████   | 806/1135 [20:11<06:59,  1.28s/it]

 71%|███████   | 807/1135 [20:13<06:49,  1.25s/it]

 71%|███████   | 808/1135 [20:14<06:43,  1.24s/it]

 71%|███████▏  | 809/1135 [20:15<06:48,  1.25s/it]

 71%|███████▏  | 810/1135 [20:16<06:39,  1.23s/it]

 71%|███████▏  | 811/1135 [20:18<06:47,  1.26s/it]

 72%|███████▏  | 812/1135 [20:19<06:41,  1.24s/it]

 72%|███████▏  | 813/1135 [20:20<06:57,  1.30s/it]

 72%|███████▏  | 814/1135 [20:22<06:50,  1.28s/it]

 72%|███████▏  | 815/1135 [20:23<06:37,  1.24s/it]

 72%|███████▏  | 816/1135 [20:24<06:32,  1.23s/it]

 72%|███████▏  | 817/1135 [20:25<06:37,  1.25s/it]

 72%|███████▏  | 818/1135 [20:26<06:40,  1.26s/it]

 72%|███████▏  | 819/1135 [20:28<06:32,  1.24s/it]

 72%|███████▏  | 820/1135 [20:29<06:30,  1.24s/it]

 72%|███████▏  | 821/1135 [20:30<06:30,  1.24s/it]

 72%|███████▏  | 822/1135 [20:31<06:22,  1.22s/it]

 73%|███████▎  | 823/1135 [20:32<06:15,  1.20s/it]

 73%|███████▎  | 824/1135 [20:34<06:09,  1.19s/it]

 73%|███████▎  | 825/1135 [20:35<06:12,  1.20s/it]

 73%|███████▎  | 826/1135 [20:36<06:08,  1.19s/it]

 73%|███████▎  | 827/1135 [20:37<06:05,  1.19s/it]

 73%|███████▎  | 828/1135 [20:39<06:16,  1.23s/it]

 73%|███████▎  | 829/1135 [20:40<06:14,  1.22s/it]

 73%|███████▎  | 830/1135 [20:41<06:06,  1.20s/it]

 73%|███████▎  | 831/1135 [20:42<06:02,  1.19s/it]

 73%|███████▎  | 832/1135 [20:43<06:01,  1.19s/it]

 73%|███████▎  | 833/1135 [20:44<06:03,  1.20s/it]

 73%|███████▎  | 834/1135 [20:46<05:58,  1.19s/it]

 74%|███████▎  | 835/1135 [20:47<06:10,  1.24s/it]

 74%|███████▎  | 836/1135 [20:48<06:02,  1.21s/it]

 74%|███████▎  | 837/1135 [20:49<05:57,  1.20s/it]

 74%|███████▍  | 838/1135 [20:50<05:53,  1.19s/it]

 74%|███████▍  | 839/1135 [20:52<05:51,  1.19s/it]

 74%|███████▍  | 840/1135 [20:53<05:47,  1.18s/it]

 74%|███████▍  | 841/1135 [20:54<05:45,  1.17s/it]

 74%|███████▍  | 842/1135 [20:55<05:42,  1.17s/it]

 74%|███████▍  | 843/1135 [20:56<05:45,  1.18s/it]

 74%|███████▍  | 844/1135 [20:58<05:44,  1.18s/it]

 74%|███████▍  | 845/1135 [20:59<05:40,  1.18s/it]

 75%|███████▍  | 846/1135 [21:00<05:37,  1.17s/it]

 75%|███████▍  | 847/1135 [21:01<05:41,  1.18s/it]

 75%|███████▍  | 848/1135 [21:02<05:57,  1.24s/it]

 75%|███████▍  | 849/1135 [21:04<05:59,  1.26s/it]

 75%|███████▍  | 850/1135 [21:05<05:49,  1.23s/it]

 75%|███████▍  | 851/1135 [21:06<05:43,  1.21s/it]

 75%|███████▌  | 852/1135 [21:08<06:02,  1.28s/it]

 75%|███████▌  | 853/1135 [21:09<05:49,  1.24s/it]

 75%|███████▌  | 854/1135 [21:10<05:46,  1.23s/it]

 75%|███████▌  | 855/1135 [21:11<05:39,  1.21s/it]

 75%|███████▌  | 856/1135 [21:12<05:36,  1.21s/it]

 76%|███████▌  | 857/1135 [21:13<05:31,  1.19s/it]

 76%|███████▌  | 858/1135 [21:15<05:28,  1.19s/it]

 76%|███████▌  | 859/1135 [21:16<05:27,  1.19s/it]

 76%|███████▌  | 860/1135 [21:17<05:25,  1.18s/it]

 76%|███████▌  | 861/1135 [21:18<05:25,  1.19s/it]

 76%|███████▌  | 862/1135 [21:19<05:22,  1.18s/it]

 76%|███████▌  | 863/1135 [21:20<05:21,  1.18s/it]

 76%|███████▌  | 864/1135 [21:22<05:19,  1.18s/it]

 76%|███████▌  | 865/1135 [21:23<05:16,  1.17s/it]

 76%|███████▋  | 866/1135 [21:24<05:15,  1.17s/it]

 76%|███████▋  | 867/1135 [21:25<05:17,  1.19s/it]

 76%|███████▋  | 868/1135 [21:26<05:25,  1.22s/it]

 77%|███████▋  | 869/1135 [21:28<06:00,  1.36s/it]

 77%|███████▋  | 870/1135 [21:29<05:48,  1.31s/it]

 77%|███████▋  | 871/1135 [21:31<05:36,  1.27s/it]

 77%|███████▋  | 872/1135 [21:32<05:31,  1.26s/it]

 77%|███████▋  | 873/1135 [21:33<05:22,  1.23s/it]

 77%|███████▋  | 874/1135 [21:35<06:18,  1.45s/it]

 77%|███████▋  | 875/1135 [21:36<05:54,  1.36s/it]

 77%|███████▋  | 876/1135 [21:37<05:38,  1.31s/it]

 77%|███████▋  | 877/1135 [21:39<05:34,  1.30s/it]

 77%|███████▋  | 878/1135 [21:40<05:46,  1.35s/it]

 77%|███████▋  | 879/1135 [21:41<05:32,  1.30s/it]

 78%|███████▊  | 880/1135 [21:42<05:32,  1.31s/it]

 78%|███████▊  | 881/1135 [21:44<05:26,  1.29s/it]

 78%|███████▊  | 882/1135 [21:45<05:19,  1.26s/it]

 78%|███████▊  | 883/1135 [21:46<05:36,  1.34s/it]

 78%|███████▊  | 884/1135 [21:48<05:34,  1.33s/it]

 78%|███████▊  | 885/1135 [21:49<05:23,  1.29s/it]

 78%|███████▊  | 886/1135 [21:50<05:12,  1.25s/it]

 78%|███████▊  | 887/1135 [21:51<05:11,  1.25s/it]

 78%|███████▊  | 888/1135 [21:53<05:07,  1.24s/it]

 78%|███████▊  | 889/1135 [21:54<05:01,  1.23s/it]

 78%|███████▊  | 890/1135 [21:55<05:03,  1.24s/it]

 79%|███████▊  | 891/1135 [21:56<04:57,  1.22s/it]

 79%|███████▊  | 892/1135 [21:57<04:55,  1.22s/it]

 79%|███████▊  | 893/1135 [21:59<04:50,  1.20s/it]

 79%|███████▉  | 894/1135 [22:00<04:48,  1.20s/it]

 79%|███████▉  | 895/1135 [22:01<04:45,  1.19s/it]

 79%|███████▉  | 896/1135 [22:02<04:41,  1.18s/it]

 79%|███████▉  | 897/1135 [22:03<04:40,  1.18s/it]

 79%|███████▉  | 898/1135 [22:05<04:41,  1.19s/it]

 79%|███████▉  | 899/1135 [22:06<04:40,  1.19s/it]

 79%|███████▉  | 900/1135 [22:07<04:37,  1.18s/it]

 79%|███████▉  | 901/1135 [22:08<04:34,  1.17s/it]

 79%|███████▉  | 902/1135 [22:09<04:35,  1.18s/it]

 80%|███████▉  | 903/1135 [22:10<04:36,  1.19s/it]

 80%|███████▉  | 904/1135 [22:12<04:33,  1.18s/it]

 80%|███████▉  | 905/1135 [22:13<04:30,  1.18s/it]

 80%|███████▉  | 906/1135 [22:14<04:29,  1.18s/it]

 80%|███████▉  | 907/1135 [22:15<04:40,  1.23s/it]

 80%|████████  | 908/1135 [22:17<04:41,  1.24s/it]

 80%|████████  | 909/1135 [22:18<04:45,  1.26s/it]

 80%|████████  | 910/1135 [22:19<04:37,  1.23s/it]

 80%|████████  | 911/1135 [22:20<04:33,  1.22s/it]

 80%|████████  | 912/1135 [22:21<04:31,  1.22s/it]

 80%|████████  | 913/1135 [22:23<04:26,  1.20s/it]

 81%|████████  | 914/1135 [22:24<04:32,  1.23s/it]

 81%|████████  | 915/1135 [22:25<04:26,  1.21s/it]

 81%|████████  | 916/1135 [22:27<04:43,  1.29s/it]

 81%|████████  | 917/1135 [22:28<04:40,  1.29s/it]

 81%|████████  | 918/1135 [22:29<04:42,  1.30s/it]

 81%|████████  | 919/1135 [22:30<04:32,  1.26s/it]

 81%|████████  | 920/1135 [22:32<04:25,  1.24s/it]

 81%|████████  | 921/1135 [22:33<04:19,  1.21s/it]

 81%|████████  | 922/1135 [22:34<04:17,  1.21s/it]

 81%|████████▏ | 923/1135 [22:35<04:23,  1.24s/it]

 81%|████████▏ | 924/1135 [22:37<04:27,  1.27s/it]

 81%|████████▏ | 925/1135 [22:38<04:22,  1.25s/it]

 82%|████████▏ | 926/1135 [22:39<04:27,  1.28s/it]

 82%|████████▏ | 927/1135 [22:41<04:48,  1.38s/it]

 82%|████████▏ | 928/1135 [22:42<05:00,  1.45s/it]

 82%|████████▏ | 929/1135 [22:44<05:07,  1.49s/it]

 82%|████████▏ | 930/1135 [22:45<04:52,  1.43s/it]

 82%|████████▏ | 931/1135 [22:46<04:41,  1.38s/it]

 82%|████████▏ | 932/1135 [22:48<04:28,  1.32s/it]

 82%|████████▏ | 933/1135 [22:49<04:17,  1.28s/it]

 82%|████████▏ | 934/1135 [22:50<04:09,  1.24s/it]

 82%|████████▏ | 935/1135 [22:51<04:06,  1.23s/it]

 82%|████████▏ | 936/1135 [22:52<04:02,  1.22s/it]

 83%|████████▎ | 937/1135 [22:54<03:58,  1.20s/it]

 83%|████████▎ | 938/1135 [22:55<03:57,  1.21s/it]

 83%|████████▎ | 939/1135 [22:56<03:53,  1.19s/it]

 83%|████████▎ | 940/1135 [22:57<03:51,  1.19s/it]

 83%|████████▎ | 941/1135 [22:58<03:49,  1.18s/it]

 83%|████████▎ | 942/1135 [23:00<03:55,  1.22s/it]

 83%|████████▎ | 943/1135 [23:01<03:54,  1.22s/it]

 83%|████████▎ | 944/1135 [23:02<03:50,  1.21s/it]

 83%|████████▎ | 945/1135 [23:03<03:54,  1.23s/it]

 83%|████████▎ | 946/1135 [23:05<03:59,  1.27s/it]

 83%|████████▎ | 947/1135 [23:06<03:58,  1.27s/it]

 84%|████████▎ | 948/1135 [23:07<03:58,  1.28s/it]

 84%|████████▎ | 949/1135 [23:08<04:00,  1.29s/it]

 84%|████████▎ | 950/1135 [23:10<03:52,  1.26s/it]

 84%|████████▍ | 951/1135 [23:11<03:46,  1.23s/it]

 84%|████████▍ | 952/1135 [23:12<03:41,  1.21s/it]

 84%|████████▍ | 953/1135 [23:13<03:40,  1.21s/it]

 84%|████████▍ | 954/1135 [23:14<03:39,  1.21s/it]

 84%|████████▍ | 955/1135 [23:16<03:36,  1.20s/it]

 84%|████████▍ | 956/1135 [23:17<03:32,  1.19s/it]

 84%|████████▍ | 957/1135 [23:18<03:31,  1.19s/it]

 84%|████████▍ | 958/1135 [23:19<03:28,  1.18s/it]

 84%|████████▍ | 959/1135 [23:20<03:27,  1.18s/it]

 85%|████████▍ | 960/1135 [23:21<03:26,  1.18s/it]

 85%|████████▍ | 961/1135 [23:23<03:24,  1.18s/it]

 85%|████████▍ | 962/1135 [23:24<03:24,  1.18s/it]

 85%|████████▍ | 963/1135 [23:25<03:22,  1.18s/it]

 85%|████████▍ | 964/1135 [23:26<03:21,  1.18s/it]

 85%|████████▌ | 965/1135 [23:27<03:23,  1.20s/it]

 85%|████████▌ | 966/1135 [23:29<03:21,  1.19s/it]

 85%|████████▌ | 967/1135 [23:30<03:18,  1.18s/it]

 85%|████████▌ | 968/1135 [23:31<03:15,  1.17s/it]

 85%|████████▌ | 969/1135 [23:32<03:15,  1.18s/it]

 85%|████████▌ | 970/1135 [23:33<03:13,  1.17s/it]

 86%|████████▌ | 971/1135 [23:35<03:15,  1.19s/it]

 86%|████████▌ | 972/1135 [23:36<03:16,  1.21s/it]

 86%|████████▌ | 973/1135 [23:37<03:13,  1.19s/it]

 86%|████████▌ | 974/1135 [23:38<03:10,  1.18s/it]

 86%|████████▌ | 975/1135 [23:39<03:07,  1.17s/it]

 86%|████████▌ | 976/1135 [23:40<03:10,  1.20s/it]

 86%|████████▌ | 977/1135 [23:42<03:07,  1.19s/it]

 86%|████████▌ | 978/1135 [23:43<03:05,  1.18s/it]

 86%|████████▋ | 979/1135 [23:44<03:03,  1.18s/it]

 86%|████████▋ | 980/1135 [23:45<03:04,  1.19s/it]

 86%|████████▋ | 981/1135 [23:46<03:05,  1.21s/it]

 87%|████████▋ | 982/1135 [23:48<03:02,  1.20s/it]

 87%|████████▋ | 983/1135 [23:49<03:04,  1.21s/it]

 87%|████████▋ | 984/1135 [23:50<03:02,  1.21s/it]

 87%|████████▋ | 985/1135 [23:51<02:59,  1.19s/it]

 87%|████████▋ | 986/1135 [23:53<03:02,  1.23s/it]

 87%|████████▋ | 987/1135 [23:54<02:59,  1.21s/it]

 87%|████████▋ | 988/1135 [23:55<02:56,  1.20s/it]

 87%|████████▋ | 989/1135 [23:56<02:54,  1.19s/it]

 87%|████████▋ | 990/1135 [23:57<02:53,  1.19s/it]

 87%|████████▋ | 991/1135 [23:58<02:51,  1.19s/it]

 87%|████████▋ | 992/1135 [24:00<02:50,  1.19s/it]

 87%|████████▋ | 993/1135 [24:01<02:47,  1.18s/it]

 88%|████████▊ | 994/1135 [24:02<02:48,  1.19s/it]

 88%|████████▊ | 995/1135 [24:03<02:46,  1.19s/it]

 88%|████████▊ | 996/1135 [24:04<02:43,  1.18s/it]

 88%|████████▊ | 997/1135 [24:05<02:42,  1.18s/it]

 88%|████████▊ | 998/1135 [24:07<02:41,  1.18s/it]

 88%|████████▊ | 999/1135 [24:08<02:41,  1.19s/it]

 88%|████████▊ | 1000/1135 [24:09<02:39,  1.18s/it]

 88%|████████▊ | 1001/1135 [24:10<02:37,  1.18s/it]

 88%|████████▊ | 1002/1135 [24:11<02:39,  1.20s/it]

 88%|████████▊ | 1003/1135 [24:13<02:36,  1.18s/it]

 88%|████████▊ | 1004/1135 [24:14<02:34,  1.18s/it]

 89%|████████▊ | 1005/1135 [24:15<02:32,  1.18s/it]

 89%|████████▊ | 1006/1135 [24:16<02:31,  1.17s/it]

 89%|████████▊ | 1007/1135 [24:17<02:30,  1.18s/it]

 89%|████████▉ | 1008/1135 [24:19<02:31,  1.20s/it]

 89%|████████▉ | 1009/1135 [24:20<02:32,  1.21s/it]

 89%|████████▉ | 1010/1135 [24:21<02:30,  1.20s/it]

 89%|████████▉ | 1011/1135 [24:22<02:29,  1.20s/it]

 89%|████████▉ | 1012/1135 [24:23<02:30,  1.22s/it]

 89%|████████▉ | 1013/1135 [24:25<02:30,  1.23s/it]

 89%|████████▉ | 1014/1135 [24:26<02:26,  1.21s/it]

 89%|████████▉ | 1015/1135 [24:27<02:27,  1.23s/it]

 90%|████████▉ | 1016/1135 [24:28<02:23,  1.21s/it]

 90%|████████▉ | 1017/1135 [24:29<02:21,  1.20s/it]

 90%|████████▉ | 1018/1135 [24:31<02:18,  1.19s/it]

 90%|████████▉ | 1019/1135 [24:32<02:18,  1.20s/it]

 90%|████████▉ | 1020/1135 [24:33<02:16,  1.19s/it]

 90%|████████▉ | 1021/1135 [24:34<02:14,  1.18s/it]

 90%|█████████ | 1022/1135 [24:35<02:15,  1.20s/it]

 90%|█████████ | 1023/1135 [24:37<02:12,  1.18s/it]

 90%|█████████ | 1024/1135 [24:38<02:11,  1.18s/it]

 90%|█████████ | 1025/1135 [24:39<02:09,  1.17s/it]

 90%|█████████ | 1026/1135 [24:40<02:09,  1.18s/it]

 90%|█████████ | 1027/1135 [24:41<02:07,  1.18s/it]

 91%|█████████ | 1028/1135 [24:42<02:06,  1.18s/it]

 91%|█████████ | 1029/1135 [24:44<02:04,  1.17s/it]

 91%|█████████ | 1030/1135 [24:45<02:04,  1.19s/it]

 91%|█████████ | 1031/1135 [24:46<02:03,  1.19s/it]

 91%|█████████ | 1032/1135 [24:47<02:01,  1.18s/it]

 91%|█████████ | 1033/1135 [24:48<01:59,  1.17s/it]

 91%|█████████ | 1034/1135 [24:50<01:58,  1.18s/it]

 91%|█████████ | 1035/1135 [24:51<01:58,  1.18s/it]

 91%|█████████▏| 1036/1135 [24:52<01:58,  1.19s/it]

 91%|█████████▏| 1037/1135 [24:53<01:55,  1.18s/it]

 91%|█████████▏| 1038/1135 [24:54<01:57,  1.21s/it]

 92%|█████████▏| 1039/1135 [24:56<01:55,  1.21s/it]

 92%|█████████▏| 1040/1135 [24:57<01:53,  1.19s/it]

 92%|█████████▏| 1041/1135 [24:58<01:50,  1.18s/it]

 92%|█████████▏| 1042/1135 [24:59<01:50,  1.19s/it]

 92%|█████████▏| 1043/1135 [25:00<01:49,  1.19s/it]

 92%|█████████▏| 1044/1135 [25:01<01:47,  1.19s/it]

 92%|█████████▏| 1045/1135 [25:03<01:46,  1.18s/it]

 92%|█████████▏| 1046/1135 [25:04<01:45,  1.18s/it]

 92%|█████████▏| 1047/1135 [25:05<01:44,  1.19s/it]

 92%|█████████▏| 1048/1135 [25:06<01:42,  1.18s/it]

 92%|█████████▏| 1049/1135 [25:07<01:40,  1.17s/it]

 93%|█████████▎| 1050/1135 [25:09<01:40,  1.18s/it]

 93%|█████████▎| 1051/1135 [25:10<01:38,  1.18s/it]

 93%|█████████▎| 1052/1135 [25:11<01:37,  1.17s/it]

 93%|█████████▎| 1053/1135 [25:12<01:36,  1.17s/it]

 93%|█████████▎| 1054/1135 [25:13<01:35,  1.18s/it]

 93%|█████████▎| 1055/1135 [25:14<01:34,  1.18s/it]

 93%|█████████▎| 1056/1135 [25:16<01:32,  1.17s/it]

 93%|█████████▎| 1057/1135 [25:17<01:30,  1.17s/it]

 93%|█████████▎| 1058/1135 [25:18<01:31,  1.19s/it]

 93%|█████████▎| 1059/1135 [25:19<01:29,  1.18s/it]

 93%|█████████▎| 1060/1135 [25:20<01:28,  1.18s/it]

 93%|█████████▎| 1061/1135 [25:21<01:26,  1.17s/it]

 94%|█████████▎| 1062/1135 [25:23<01:26,  1.19s/it]

 94%|█████████▎| 1063/1135 [25:24<01:25,  1.18s/it]

 94%|█████████▎| 1064/1135 [25:25<01:23,  1.18s/it]

 94%|█████████▍| 1065/1135 [25:26<01:22,  1.18s/it]

 94%|█████████▍| 1066/1135 [25:27<01:21,  1.19s/it]

 94%|█████████▍| 1067/1135 [25:29<01:20,  1.19s/it]

 94%|█████████▍| 1068/1135 [25:30<01:18,  1.18s/it]

 94%|█████████▍| 1069/1135 [25:31<01:17,  1.17s/it]

 94%|█████████▍| 1070/1135 [25:32<01:17,  1.19s/it]

 94%|█████████▍| 1071/1135 [25:33<01:15,  1.19s/it]

 94%|█████████▍| 1072/1135 [25:35<01:14,  1.19s/it]

 95%|█████████▍| 1073/1135 [25:36<01:13,  1.18s/it]

 95%|█████████▍| 1074/1135 [25:37<01:18,  1.28s/it]

 95%|█████████▍| 1075/1135 [25:38<01:15,  1.27s/it]

 95%|█████████▍| 1076/1135 [25:40<01:12,  1.24s/it]

 95%|█████████▍| 1077/1135 [25:41<01:10,  1.22s/it]

 95%|█████████▍| 1078/1135 [25:42<01:08,  1.20s/it]

 95%|█████████▌| 1079/1135 [25:43<01:07,  1.21s/it]

 95%|█████████▌| 1080/1135 [25:44<01:06,  1.21s/it]

 95%|█████████▌| 1081/1135 [25:46<01:04,  1.19s/it]

 95%|█████████▌| 1082/1135 [25:47<01:02,  1.19s/it]

 95%|█████████▌| 1083/1135 [25:48<01:02,  1.21s/it]

 96%|█████████▌| 1084/1135 [25:49<01:00,  1.20s/it]

 96%|█████████▌| 1085/1135 [25:50<01:01,  1.23s/it]

 96%|█████████▌| 1086/1135 [25:52<01:00,  1.23s/it]

 96%|█████████▌| 1087/1135 [25:53<00:58,  1.21s/it]

 96%|█████████▌| 1088/1135 [25:54<00:56,  1.19s/it]

 96%|█████████▌| 1089/1135 [25:55<00:55,  1.20s/it]

 96%|█████████▌| 1090/1135 [25:56<00:53,  1.20s/it]

 96%|█████████▌| 1091/1135 [25:58<00:52,  1.20s/it]

 96%|█████████▌| 1092/1135 [25:59<00:51,  1.19s/it]

 96%|█████████▋| 1093/1135 [26:00<00:49,  1.18s/it]

 96%|█████████▋| 1094/1135 [26:01<00:48,  1.18s/it]

 96%|█████████▋| 1095/1135 [26:02<00:47,  1.20s/it]

 97%|█████████▋| 1096/1135 [26:04<00:47,  1.22s/it]

 97%|█████████▋| 1097/1135 [26:05<00:46,  1.22s/it]

retrying after error: 


 97%|█████████▋| 1098/1135 [27:41<18:22, 29.79s/it]

 97%|█████████▋| 1099/1135 [27:42<12:43, 21.20s/it]

 97%|█████████▋| 1100/1135 [27:44<08:51, 15.19s/it]

 97%|█████████▋| 1101/1135 [27:45<06:15, 11.03s/it]

 97%|█████████▋| 1102/1135 [27:46<04:26,  8.07s/it]

 97%|█████████▋| 1103/1135 [27:47<03:12,  6.00s/it]

 97%|█████████▋| 1104/1135 [27:48<02:21,  4.56s/it]

 97%|█████████▋| 1105/1135 [27:50<01:46,  3.54s/it]

 97%|█████████▋| 1106/1135 [27:51<01:22,  2.84s/it]

 98%|█████████▊| 1107/1135 [27:52<01:05,  2.34s/it]

 98%|█████████▊| 1108/1135 [27:53<00:53,  1.99s/it]

 98%|█████████▊| 1109/1135 [27:54<00:45,  1.76s/it]

 98%|█████████▊| 1110/1135 [27:56<00:40,  1.62s/it]

 98%|█████████▊| 1111/1135 [27:57<00:35,  1.48s/it]

 98%|█████████▊| 1112/1135 [27:58<00:31,  1.39s/it]

 98%|█████████▊| 1113/1135 [27:59<00:29,  1.33s/it]

 98%|█████████▊| 1114/1135 [28:00<00:27,  1.29s/it]

 98%|█████████▊| 1115/1135 [28:02<00:25,  1.25s/it]

 98%|█████████▊| 1116/1135 [28:03<00:23,  1.22s/it]

 98%|█████████▊| 1117/1135 [28:04<00:21,  1.20s/it]

 99%|█████████▊| 1118/1135 [28:05<00:20,  1.20s/it]

 99%|█████████▊| 1119/1135 [28:06<00:19,  1.20s/it]

 99%|█████████▊| 1120/1135 [28:07<00:17,  1.20s/it]

 99%|█████████▉| 1121/1135 [28:09<00:16,  1.19s/it]

 99%|█████████▉| 1122/1135 [28:10<00:15,  1.19s/it]

 99%|█████████▉| 1123/1135 [28:11<00:14,  1.19s/it]

 99%|█████████▉| 1124/1135 [28:12<00:13,  1.19s/it]

 99%|█████████▉| 1125/1135 [28:13<00:12,  1.20s/it]

 99%|█████████▉| 1126/1135 [28:15<00:11,  1.27s/it]

 99%|█████████▉| 1127/1135 [28:16<00:09,  1.24s/it]

 99%|█████████▉| 1128/1135 [28:17<00:08,  1.26s/it]

 99%|█████████▉| 1129/1135 [28:19<00:07,  1.23s/it]

100%|█████████▉| 1130/1135 [28:20<00:06,  1.21s/it]

100%|█████████▉| 1131/1135 [28:21<00:04,  1.20s/it]

100%|█████████▉| 1132/1135 [28:22<00:03,  1.19s/it]

100%|█████████▉| 1133/1135 [28:23<00:02,  1.19s/it]

100%|█████████▉| 1134/1135 [28:24<00:01,  1.19s/it]

100%|██████████| 1135/1135 [28:25<00:00,  1.16s/it]

100%|██████████| 1135/1135 [28:25<00:00,  1.50s/it]

retrieved 45985 of 46007 unique commits; 22 not found in WoC commit.tch in this run


## Step 3: write `rkabarwa_project_summary.csv` (semicolon separated)

In [4]:
dfc = df.merge(pd.DataFrame(done.values()), left_on='sha1', right_on='commit')
summary = dfc[['project', 'sha1', 'author', 'time', 'message']]
summary.columns = ['project_wocid', 'commit_sha1', 'author', 'time', 'commit message']
summary.to_csv(netid + '_project_summary.csv', sep=';', index=False)
print(len(summary), 'rows written')
summary.head()

45985 rows written


,project_wocid,commit_sha1,author,time,commit message
0,dlr-rm_blenderproc,0003f0a70b91315ba228f3adf329bf06cf91641a,Klaus Strobl <44800028+5trobl@users.noreply.gi...,1676903306,chore(readme): Update links
1,dlr-rm_blenderproc,001167c20e8727ad888d68483dd087196a9fc6d1,Maximilian Denninger <maximilian.denninger@dlr...,1587566621,feat(PartSphere): add a vector to specify the ...
2,dlr-rm_blenderproc,0012deb4b57bf2d5a967693bf04757ee0915cb86,Dmitry Olefir <dmitry.olefir@dlr.de>,1579791599,example refactoring\n
3,dlr-rm_blenderproc,001d788bf307f82225a4d20643c1ba216fa2c028,Maximilian Denninger <maximilian.denninger@dlr...,1587550025,doc(ShapeNet/README): add proper README for ba...
4,dlr-rm_blenderproc,0022f326761cbe375582bebc2d36934ec182cde4,Martin Sundermeyer <martin.sundermeyer@dlr.de>,1590422965,started elevation uniform\n


## Step 4: WoC stats per project

In [5]:
stats = summary.groupby('project_wocid').agg(
    ncommits=('commit_sha1', 'count'),
    nauthors=('author', 'nunique'),
    min_time=('time', 'min'),
    max_time=('time', 'max'))
stats['min_time'] = pd.to_datetime(stats.min_time, unit='s')
stats['max_time'] = pd.to_datetime(stats.max_time, unit='s')
stats = stats.reindex(projects.WoC)
print(stats.to_markdown())

| WoC                                             |   ncommits |   nauthors | min_time            | max_time            |
|:------------------------------------------------|-----------:|-----------:|:--------------------|:--------------------|
| dlr-rm_blenderproc                              |       5512 |        152 | 2019-06-28 09:32:06 | 2026-03-15 01:03:33 |
| riatelab_cartography                            |        737 |         20 | 2015-06-11 08:27:11 | 2025-07-23 07:06:59 |
| patrickzib_sfa                                  |        466 |         13 | 2016-02-22 14:38:46 | 2025-05-02 13:28:17 |
| openexoplanetcatalogue_open_exoplanet_catalogue |       7241 |         90 | 2011-01-11 03:03:32 | 2025-11-05 07:01:15 |
| snakemake-workflows_rna-seq-kallisto-sleuth     |       1420 |         47 | 2018-12-12 10:51:17 | 2025-10-31 11:53:49 |
| brianmanderson_dicom_utilities                  |        407 |         14 | 2019-10-09 18:50:49 | 2021-07-22 19:01:38 |
| iraf-community_iraf   

## Step 5: GitHub stats (stars, forks, last commit date)

In [6]:
gh = []
for wid, url in zip(projects.WoC, projects.GH):
    repo = url.replace('https://github.com/', '')
    r = requests.get(f'https://api.github.com/repos/{repo}').json()
    c = requests.get(f'https://api.github.com/repos/{repo}/commits?per_page=1').json()
    gh.append({'project': wid, 'repo': repo,
               'nstars': r['stargazers_count'], 'nforks': r['forks_count'],
               'lastGHCommitDate': c[0]['commit']['committer']['date'][:10]})
    time.sleep(1)
gh = pd.DataFrame(gh)
print(gh.to_markdown(index=False))

| project                                         | repo                                            |   nstars |   nforks | lastGHCommitDate   |
|:------------------------------------------------|:------------------------------------------------|---------:|---------:|:-------------------|
| dlr-rm_blenderproc                              | DLR-RM/BlenderProc                              |     3730 |      519 | 2026-01-07         |
| riatelab_cartography                            | riatelab/cartography                            |      403 |       44 | 2025-07-23         |
| patrickzib_sfa                                  | patrickzib/SFA                                  |      321 |       67 | 2025-05-02         |
| openexoplanetcatalogue_open_exoplanet_catalogue | OpenExoplanetCatalogue/open_exoplanet_catalogue |     1128 |      224 | 2026-09-28         |
| snakemake-workflows_rna-seq-kallisto-sleuth     | snakemake-workflows/rna-seq-kallisto-sleuth     |       69 |       48 | 2026-0

## Part 1 results

WoC columns come from the retrieved commits (author time, UTC); GitHub columns were collected on 2026-09-28.

| Project | WoC commits | WoC authors | Min time | Max time | GitHub stars | GitHub forks | Last GitHub commit |
|---|---:|---:|---|---|---:|---:|---|
| dlr-rm_blenderproc | 5,512 | 152 | 2019-06-28 | 2026-03-15 | 3,730 | 519 | 2026-01-07 |
| riatelab_cartography | 737 | 20 | 2015-06-11 | 2025-07-23 | 403 | 44 | 2025-07-23 |
| patrickzib_sfa | 466 | 13 | 2016-02-22 | 2025-05-02 | 321 | 67 | 2025-05-02 |
| openexoplanetcatalogue_open_exoplanet_catalogue | 7,241 | 90 | 2011-01-11 | 2025-11-05 | 1,128 | 224 | 2026-09-28 |
| snakemake-workflows_rna-seq-kallisto-sleuth | 1,420 | 47 | 2018-12-12 | 2025-10-31 | 69 | 48 | 2026-08-17 |
| brianmanderson_dicom_utilities | 407 | 14 | 2019-10-09 | 2021-07-22 | 108 | 32 | 2026-09-08 |
| iraf-community_iraf | 2,782 | 36 | 1989-02-22 | 2025-11-02 | 180 | 54 | 2026-09-25 |
| dgerosa_precession | 1,191 | 18 | 2015-11-12 | 2025-08-26 | 42 | 18 | 2026-06-06 |
| plasmapy_plasmapy | 25,686 | 355 | 2015-11-24 | 2026-04-17 | 714 | 374 | 2026-09-28 |
| iqusoft_intel-qs | 543 | 58 | 2017-11-06 | 2025-09-22 | 264 | 76 | 2026-09-11 |